cargo/deprecation_warning_test.go

```diff
@@ -0,0 +1,86 @@
+package cargo_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/packit/cargo"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testDeprecationWarning(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect       = NewWithT(t).Expect
+		buildpackDir string
+
+		deprecationWarning cargo.DeprecationWarning
+	)
+	it.Before(func() {
+		var err error
+		buildpackDir, err = ioutil.TempDir("", "buildpack")
+		Expect(err).NotTo(HaveOccurred())
+
+		deprecationWarning = cargo.NewDeprecationWarning()
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(buildpackDir)).To(Succeed())
+	})
+
+	context("WarnDepercatedFields", func() {
```

Comment:
I noticed "deprecated" misspelled "depercated" throughout this file, which isn't a huge deal but if you're fixing other spellings might as well do it all